src/VecSim/vec_sim_common.h

```diff
@@ -26,6 +26,26 @@ typedef enum { VecSimAlgo_BF, VecSimAlgo_HNSWLIB } VecSimAlgo;
 // Distance metric
 typedef enum { VecSimMetric_L2, VecSimMetric_IP, VecSimMetric_Cosine } VecSimMetric;
 
+/**
+ * @brief Query Runtime raw parameters.
+ * Use VecSimIndex_ResolveParams to generate VecSimQueryParams from array of VecSimRawParams.
+ *
+ */
+typedef struct {
+    const char *name;
+    size_t nameLen;
+    const char *value;
+    size_t valLen;
+} VecSimRawParam;
+
+typedef enum {
+    VecSimErr_OK = 0,
+    VecSimErr_MissingParamStruct,
+    VecSimErr_AlreadySet,
+    VecSimErr_UnknownParam,
+    VecSimErr_BadValue
+} VecSimResolveCode;
```

Comment:
This is for the specific param resolve use case, right? if so:
1. please us specific param resolve prefix like `VecSimParamResolverErr_BadValue`
2. don't use contradicting words in the same code `VecSimErr_Ok`
3. take the above error code and have it as a define, so if in the future we will need additional error code usage, we will have `VecSim_OK` shared